pre_block.go

```diff
@@ -0,0 +1,21 @@
+package dbft
+
+// PreBlock is a generic interface for a preBlock used by anti-MEV dBFT extension.
+type PreBlock[H Hash] interface {
+	// Hash returns PreBlock hash.
+	Hash() H // needed for informational log, but may be removed.
+
+	// Data returns PreBlock's data CNs need to exchange during Commit phase.
+	// It's not a final block signature.
+	Data() []byte // required
+	// SetData generates and sets PreBlock's data CNs need to exchange during Commit
+	// phase.
+	SetData(key PrivateKey) error // required
```

Comment:
Getters/setters?

Reply:
What exactly do you want? `GetData` instead of `Data`?

Reply:
Usually getters/setters mean that something is wrong with the interface. Likely `SetData` can be avoided.

Reply:
Then it’s wrong with the old implementation because the logic is the same as for our old Commit.Sign() and Commit.Signature().